lib/ansible/plugins/netconf/__init__.py

```diff
@@ -101,6 +104,11 @@ class NetconfBase(AnsiblePlugin):
     __rpc__ = ['get_config', 'edit_config', 'get_capabilities', 'get']
 
     def __init__(self, connection):
+        if not HAS_NCCLIENT:
+            raise AnsibleError(
+                'The required "ncclient" python library is required to use the netconf connection type: %s.\n'
+                'Please run pip install ncclient' % to_native(NCCLIENT_IMP_ERR)
+            )
```

Comment:
i recommend you use `missing_required_lib` from module_utils.basic as it adds 'hostname' and 'which python' to avoid confusion

Reply:
isn't __init__ too early still? normally we return the error when we attempt to use the plugin, since they can be instantiated for other reasons (i.e doc reading/config/etc)